treeherder/model/bug_suggestions.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import re
+import urllib
+import json
+
+from django.core.urlresolvers import reverse
+from django.conf import settings
+
+from thclient import TreeherderArtifactCollection
+from treeherder.etl import th_publisher
+
+logger = logging.getLogger(__name__)
+
+
+def generate_bug_suggestions_artifacts(project, artifact_list):
+    """
+    Create bug suggestions artifact(s) for any text_log_summary artifacts.
+
+    ``artifact_list`` here is a list of artifacts that may contain one or more
+        ``text_log_artifact`` objects.  If it does, we extract the error lines
+        from it.  If there ARE error lines, then we generate the
+        ``bug suggestions`` artifact from them and post it.
+    """
+
+    tac = TreeherderArtifactCollection()
+
+    for artifact in artifact_list:
+        # this is the only artifact name eligible to trigger generation of bug suggestions.
+        if artifact['name'] != 'text_log_summary':
+            continue
+
+        all_errors = get_all_errors(artifact)
+        if all_errors:
+            ta = tac.get_artifact({
+                "job_guid": artifact['job_guid'],
+                "name": 'Bug suggestions',
+                "type": 'json',
+                "blob": json.dumps(get_bug_suggestions(all_errors))
+            })
+            tac.add(ta)
+
+    if tac.get_collection_data():
+        th_publisher.post_treeherder_collections({project: tac})
+
+
+def get_all_errors(artifact):
+    """Extract the error lines from an artifact's blob field"""
+
+    artifact_blob = json.loads(artifact['blob'])
+    if isinstance(artifact_blob, dict):
+        return artifact_blob.get('step_data', {}).get('all_errors', [])
+
+
```

Comment:
end of the new code.  From here down was pretty much just what I copied from the ``log_parser/utils`` file.  This first function was extracted from the previous ``extract_text_log_artifacts`` function.  Just removed the lines that dealt with the ``ArtifactBuilderCollection``.  Otherwise verbatim.